### PR TITLE
Adding precommit hooks

### PR DESCRIPTION
**Pull request by AnomalRoil**
Since @GlacierWalrus requested these...

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    -   id: check-added-large-files
+-   repo: https://github.com/golangci/golangci-lint
+    rev: v1.58.1
+    hooks:
+    -   id: golangci-lint
+- repo: local
+  hooks:
+    - id: go-test-short
+      name: go-test-short
+      entry: bash -c 'go test -tags conn_insecure -short ./...'
```
Comment (GlacierWalrus) [inline .pre-commit-config.yaml:19]: You run tests on each commit?

Comment (AnomalRoil) [reply .pre-commit-config.yaml:19]: every commit should compile lest we can't `git bissect` on it.

Comment (laithalissa) [reply .pre-commit-config.yaml:19]: sure but we don't merge things which don't pass the full build, and we squash

Comment (AnomalRoil) [reply .pre-commit-config.yaml:19]: I guess it'd be best to change it to do that on push only yeah.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  rev: v1.59.1
+  hooks:
+    - id: golangci-lint
+      stages: [pre-push]
+- repo: local
+  hooks:
+    - id: go-test-short
+      stages: [pre-push]
+      name: go-test-short
+      entry: bash -c 'go test -tags conn_insecure -short ./...'
```
Comment (GlacierWalrus) [inline .pre-commit-config.yaml:30]: Do you think it's worth adding `go fmt` as either a pre-commit or pre-push?

**Comment by GlacierWalrus** [comment]
you've included extra changes it seems

File: .pre-commit-config.yaml (hunk 2 above)
Comment (CluEleSsUK) [reply .pre-commit-config.yaml:30]: we have `golangci-lint` above - I think that enforces format already

**Review by CluEleSsUK — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (GlacierWalrus) [reply .pre-commit-config.yaml:30]: I mean a writing formatter, rather than just a linter. Unless you mean golangci-lint alone applies formatting already

Comment (CluEleSsUK) [reply .pre-commit-config.yaml:30]: I believe it does (with —fix anyway)

Comment (GlacierWalrus) [reply .pre-commit-config.yaml:30]: yeah so it doesn't atm, would you prefer it does automatically or not? I'd much rather it fails but does the formatting than just fails and makes me run the command 🤷

Comment (CluEleSsUK) [reply .pre-commit-config.yaml:30]: well the issue is it‘d have to commit the changes. That’s fine, though might need a separate action (or one of the magic bots)

**Comment by GlacierWalrus** [comment]
fine fine, merge it, we can iron it out later

**Review by GlacierWalrus — APPROVED** [review]

**Pull request merged**